### PR TITLE
Disable default bash xtrace and fix but in variables

### PR DESCRIPTION
**Pull request by omriiluz**
Two changes in this pull request -
1. disable default bash xtrace with option to enable with DEBUG environment variable.
2. fix a bug that caused accumulation of routes and push directives when ovpn_genconfig was used multiple times.

### CODE REVIEW

**Comment by kylemanna** [comment]
This pull request has two features.  The disabling of the debug feature adds no value and adds complexity to the code.  The debugging output is harmless and captured by docker log or printed to stderr and easily discarded.  Why?

As for the push/routes arrays, only part of that makes sense to me.

File: bin/ovpn_genconfig [hunk 1]
```diff
@@ -104,6 +110,11 @@ while getopts ":r:s:du:cp:DNm:" opt; do
     esac
 done
 
+# if new routes were not defined with -r, use default
+[ ${#TMP_ROUTES[@]} -gt 0 ] && OVPN_ROUTES=("${TMP_ROUTES[@]}")
```
Comment (kylemanna) [inline bin/ovpn_genconfig]: Why do this if it happens on line 123/134 anyways?  I don't see what this fixes.

File: bin/ovpn_genconfig [hunk 2]
```diff
@@ -104,6 +110,11 @@ while getopts ":r:s:du:cp:DNm:" opt; do
...
+
+# if new push directives were not defined with -p, use default
+[ ${#TMP_PUSH[@]} -gt 0 ] && OVPN_PUSH=("${TMP_PUSH[@]}")
```
Comment (kylemanna) [inline bin/ovpn_genconfig]: I understand the need for this, but why not implement it like OVPN_ROUTES was previously line 123/134?

**Comment by kylemanna** [comment]
> As for the push/routes arrays, only part of that makes sense to me.

Never mind, I get it now.  If you want to rebase this branch without the debug commit I'll happily merge it. :+1:

**Comment by omriiluz** [comment]
Sorry for bundling two thing in one PR, should have known better. 

As for the debug part - we use genconfig as part as our deploy scripts and the debug makes the console messy for no reason. I can filter stderr but then I'll list real errors going to stderr. 

If you still feel it's unnecessary I'll eliminate it and resubmit. But if you don't mind this little change it would be great to add. Let me know.

**Comment by kylemanna** [comment]
Spent some time thinking about it,  I suppose the debug info was probably only useful to me during early development.  Thanks!

**Comment by kylemanna** [comment]
@omriiluz btw, the intent is for the return code (and `set -e` by association) to propagate the occurrence of an error to the caller.  If you're running in to issues deploying nodes where this isn't the case then we should update the scripts to report an error.

**Pull request merged**